lmnet/executor/train.py

```diff
@@ -58,6 +58,13 @@ def setup_dataset(config, subset, rank):
     return DatasetIterator(dataset, seed=rank, enable_prefetch=enable_prefetch)
 
 
+def close_dataset(dataset):
```

Comment:
@joelN123 
Hi, thank you for try to fix the bug!!
`train.py` is the main loop of training.
It's better to implement `close_dataset` feature in `DatasetItretaor` class.
How about that?

Reply:
ah, right, as a method of datasetiterator is better, yes.

Reply:
I will make a change now and test it still is working. Thank you for the advice

Reply:
oh, I should also remove the == True in the if statement

Reply:
I made that change, and it runs fine (hangs up properly) still.